src/test/java/org/javarosa/entities/EntityFormParserTest.java

```diff
@@ -1,90 +1,41 @@
-            ),
-            head(
-                title("Create entity form"),
-                model(
-                    mainInstance(
-                        t("data id=\"create-entity-form\"",
-                            t("name"),
-                            t("meta",
-                                t("entity dataset=\"people\" create=\"true\"")
-                            )
-                        )
-                    ),
-                    bind("/data/name").type("string").withAttribute("entities", "saveto", "name")
-                )
-            ),
-            body(
-                input("/data/name")
-            )
-        );
-
-        XFormParser parser = new XFormParser(new InputStreamReader(new ByteArrayInputStream(form.asXml().getBytes())));
-        FormDef formDef = parser.parse(null);
+    public void parseAction_findsCreateWithTrueString() {
+        TreeElement entityElement = new TreeElement("entity");
+        entityElement.setAttribute(null, "create", "true");
 
-        EntityAction action = EntityFormParser.parseAction(EntityFormParser.getEntityElement(formDef.getMainInstance()));
+        EntityAction action = EntityFormParser.parseAction(entityElement);
         assertThat(action, equalTo(EntityAction.CREATE));
     }
 
     @Test
-    public void parseAction_findsUpdateWithTrueString() throws XFormParser.ParseException {
-        XFormsElement form = XFormsElement.html(
-            asList(
-                new Pair<>("entities", "http://www.opendatakit.org/xforms/entities")
-            ),
-            head(
-                title("Create entity form"),
-                model(
-                    mainInstance(
-                        t("data id=\"create-entity-form\"",
-                            t("name"),
-                            t("meta",
-                                t("entity dataset=\"people\" update=\"true\"")
-                            )
-                        )
-                    ),
-                    bind("/data/name").type("string").withAttribute("entities", "saveto", "name")
-                )
-            ),
-            body(
-                input("/data/name")
-            )
-        );
-
-        XFormParser parser = new XFormParser(new InputStreamReader(new ByteArrayInputStream(form.asXml().getBytes())));
-        FormDef formDef = parser.parse(null);
+    public void parseAction_findsUpdateWithTrueString() {
+        TreeElement entityElement = new TreeElement("entity");
+        entityElement.setAttribute(null, "update", "true");
 
-        EntityAction dataset = EntityFormParser.parseAction(EntityFormParser.getEntityElement(formDef.getMainInstance()));
+        EntityAction dataset = EntityFormParser.parseAction(entityElement);
         assertThat(dataset, equalTo(EntityAction.UPDATE));
     }
+
+    @Test
+    public void parseLabel_whenLabelIsAnInt_convertsToString() {
+        TreeElement labelElement = new TreeElement("label");
```

Comment:
I think we should have consts for values like this one `label`  and maybe for some others too like `id`